geo-types/src/arbitrary.rs

```diff
@@ -0,0 +1,131 @@
+use crate::{
+    CoordFloat, Coordinate, Geometry, GeometryCollection, LineString, MultiLineString, MultiPoint,
+    MultiPolygon, Point, Polygon, Rect, Triangle,
+};
+
+impl<'a, T: arbitrary::Arbitrary<'a> + CoordFloat> arbitrary::Arbitrary<'a>
+    for Coordinate<T>
+{
+    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+        let x = u.arbitrary::<T>()?;
+        if x.is_nan() {
+            return Err(arbitrary::Error::IncorrectFormat);
+        }
+
+        let y = u.arbitrary::<T>()?;
+        if y.is_nan() {
+            return Err(arbitrary::Error::IncorrectFormat);
+        }
+
+        Ok(Coordinate { x, y })
+    }
+}
+
+impl<'a, T: arbitrary::Arbitrary<'a> + CoordFloat> arbitrary::Arbitrary<'a>
+    for Point<T>
+{
+    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+        u.arbitrary::<Coordinate<T>>().map(Point)
+    }
+}
+
+impl<'a, T: arbitrary::Arbitrary<'a> + CoordFloat> arbitrary::Arbitrary<'a>
+    for LineString<T>
+{
+    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+        let coords = u.arbitrary::<Vec<Coordinate<T>>>()?;
+
+        if coords.len() < 2 {
+            return Err(arbitrary::Error::IncorrectFormat);
```

Comment:
I'm not familiar with `arbitrary`, but it seems like, to ensure we construct a valid linestring, we do a sort of "guess and check" strategy. 

If the guessed vector is less than 2 coords, we return an error, and presumably(?), we'll try again until a Vec of at least length 2 is created.

Is that about right?

Did you look at https://docs.rs/arbitrary/0.4.7/arbitrary/trait.Arbitrary.html#method.size_hint - it seems like it exists to avoid this guess-and-check approach if I'm understanding.

Reply:
Great great catch. Addressed in https://github.com/georust/geo/pull/622/commits/1c4399fd4536d799cdf6740d385826f122f0d0da. I think I did that right...

Reply:
Scratch that, maybe https://github.com/georust/geo/pull/622/commits/a7ad921307846fd4ad1358234fe604d900ff35c3 makes more sense

Reply:
Actually no we definitely don't want an upper bound: https://github.com/georust/geo/pull/622/commits/c2a3a4a21160a2604fd367725634e549250b26cb

I promise I will stop force pushing now :)

Reply:
Ugh what am I doing `t` is not a variable here. Going back to https://github.com/georust/geo/pull/622/commits/c3811870d5835d4a7606af314e4769efa2cb3bfc

git_commit_amend_force_push2_last_FINAL_FINAL.doc